app/admin/debate.rb

```diff
@@ -56,7 +56,7 @@
   end
 
   member_action :code, method: :post do
-    resource.generate_code
+    resource.update!(code: CodeGenerator.new.generate)
```

Comment:
* This will raise error if code is already used (there is unique constraint id db).
* In debate maker you are instantiating `CodeGenerator` with configuration values and here you depend on defaults, this can lead to different code schema being generated when defaults will change.